### PR TITLE
Notation: move rotations under 12a and attempt to describe valid FMC solutions as clearly as possible.

### PR DESCRIPTION
**Pull request by lgarron**
The new examples for E2c++ should help with concerns from issue #456.
The change to Article 12 also solves the (arguable) issue described in issue #546.

@Laura-O, how do these changes look to you?

### CODE REVIEW

File: wca-regulations.md [hunk 1]
```diff
@@ -493,11 +493,11 @@ Note: Because Article and Regulation numbers are not reassigned when Regulations
...
     - E2b) All competitors have a total time limit of 60 minutes to devise a solution.
         - E2b1) The judge should call "5 MINUTES REMAINING" at 55 minutes, and must call "STOP" at 60 minutes.
-    - E2c) At 60 minutes, each competitor must give the judge a legibly written solution with the competitor's name, using the notation defined for Outer Block Turn Metric (described in [Regulation 12a](regulations:regulation:12a)). Penalty: disqualification of the attempt (DNF).
-    - E2d) The length of the solution is calculated in Outer Block Turn Metric (see [Regulation 12a](regulations:regulation:12a)).
-        - E2d1) The competitor is permitted a maximum solution length of 80 (moves and rotations).
+    - E2c) At 60 minutes, each competitor must give the judge a legibly written solution with the competitor's name using the notation for NxNxN puzzles (described in [Regulation 12a](regulations:regulation:12a)), such that starting with a solved puzzle and applying the scramble sequence followed by the solution results in a solved puzzle. Penalty: disqualification of the attempt (DNF).
```
Comment (lgarron) [inline wca-regulations.md]: An unsubmitted or invalid solution is clearly a DNF. Have we historically allowed leniency for missing names if the competitor can be clearly identified after the fact? (I don't recall recent incidents.)

Comment (lgarron) [reply wca-regulations.md]: I've added a draft solution in E2c1+) at https://github.com/thewca/wca-regulations/pull/575/files#diff-8356179f2de288bb7a0aa8a610b9a487R208 that balances practical leniency (you can write your name in front of the judge while submitting your solution) with a clearly enforceable line (you can't add it any later).

File: wca-regulations.md [hunk 2]
```diff
@@ -491,13 +492,18 @@ Note: Because Article and Regulation numbers are not reassigned when Regulations
...
+    - E2d) The competitor's result is the number of moves in their solution, calculated using Outer Block Turn Metric (see [Regulation 12a5](regulations:regulation:12a5)).
+        - E2d1) The competitor's solution must not exceed 80 moves (including rotations) when calculated using Execution Turn Metric (see [Regulation 12a6](regulations:regulation:12a6)). Penalty: disqualification of the attempt (DNF).
     - E2e) The competitor's solution must not be directly derived from any part of the scramble sequence. Penalty: disqualification of the attempt (DNF), at the discretion of the WCA Delegate.
-        - E2e1) The WCA Delegate may ask the competitor to explain the purpose of each move in their solution, irrespective of scrambling algorithm. If the competitor cannot give a valid explanation, the attempt is disqualified (DNF).
+        - E2e1) The WCA Delegate may ask the competitor to explain the purpose of each move in their solution, irrespective of the scrambled sequence. If the competitor cannot give a valid explanation, the attempt is disqualified (DNF).
```
Comment (jfly) [inline wca-regulations.md]: "scrambled sequence" -> "scramble sequence"?

**Review by jfly — COMMENTED** [review]
This all looks like an improvement to me!

**Comment by lgarron** [comment]
Another approach is to require competitors to submit their solutions on FMC sheets, which encourage unambiguous formatting. Competitions usually print FMC sheets, and competitors are used to them, so this is fairly reasonable.

However, requiring FMC sheets means one of two possibilities:

### Every competitor is allowed only one FMC sheet
Tis requires them to juggle backup solutions and will probably lead to competitors transcribing their solutions under pressure in the last few minutes. It might lead to more incidents of competitors scribbling just a few seconds past the time limit.

### Competitors are allowed to ask for extra sheets (possibly up to a limit)
This requires organizers to print out more sheets. 

If a competitor asks for a sheet they're entitled to – but there aren't enough – then we presumably need to fall back to allowing solutions on any paper.  Even though this might lead to fewer incidents, we would still need clear Regulations for how solutions on spare paper are formatted.

However, it might be good enough to keep plenty of blank solution sheets (which TNoodle already generates) on hand. Those can be kept by organizers across competitions, so it's more reasonable to ask organizers to have a lot of them at any given competition.

File: wca-regulations.md (hunk 2 above)
Comment (lgarron) [reply wca-regulations.md]: Fixed, thanks!

**Comment by Baiqiang** [comment]
The clarification E2c4+ makes me recall the 19 moves of Vladislav Ushakov. I've argued that the solution contained xyz in () but some thought the () could be ignored.

**Comment by lgarron** [comment]
> The clarification E2c4+ makes me recall the 19 moves of Vladislav Ushakov. I've argued that the solution contained xyz in () but some thought the () could be ignored.

Indeed, one purpose of this PR is to clarify that parentheses are not valid notation.

(The only reason E2c4+ doesn't cover them is because it seemed so obvious that it wasn't worth mentioning. Incident discussions have proven otherwise.)

**Comment by Laura-O** [comment]
There are two main effects requiring solutions to be on official sheets would have:
1. Competitors will have to rewrite their solutions more often (for example because they forgot to write down moves and are not allowed to add them as extra moves).
2. Competitors have less space to write down their solution (only the main, official sheet).

I think this change in combination with a stricter handling of the notation does change the event significantly.

Apart from this, it also adds a potential unfair treatment of competitors: let's say, 3 minutes before the end of the attempt, several competitors ask for an additional sheet and the judge is not fast enough to give everyone a sheet. One competitor manages to write down the solution in time, others don't because they received the sheet later. I think this generates some randomness which we do not want to deal with.

Additionally, I think the descirption of allowed notation is ambiguous. These are two solutions which I found when looking through about 10 of the solution sheets from a competition I delegates and it's already not clear to me, how they would be judged with this PR:

img_20180725_210904

There is no star or arrow, but it's not written in-line and it's an extra move. If that would be allowed, what if there was a small line going from the crossed out move to the replacement move?

What about this one?
img_20180725_212737

Is this a DNF because the competitor drew an arrow and it's an extra move? Or is it OK because it's basically a move to be done in the written order, and one could say it's in-line?

Edit: So to summarize my thoughts on this: I think it will generate a ton of decisions which are handled differently by different delegates, which is making the situation worse, not better.

**Comment by lgarron** [comment]
Well, I designed this PR specifically so that examples like yours are clear DNFs.

First image: T/F is an ambiguous symbol, R is not written in line, and the struck-through move is not clearly blacked out (although the latter is at the discretion of the Delegate).

Second image: Even if it's not an arrow, the bracey thing is not a valid symbol.

Do you think these cases are actually ambiguous under this revision?

**Comment by pedrosino** [comment]
I think both examples Laura gave should remain valid. I also think requiring that the solution is delivered in the "official" sheet is probably overcomplicating, and maybe a kickback.

As for the struck-through move, it's clearly blacked out to me, and I would only consider the one on the top.
Requiring that competitors re-write the whole thing if they find a move missing after it's all done is not fair in my opinion.

**Comment by lgarron** [comment]
> Requiring that competitors re-write the whole thing if they find a move missing after it's all done is not fair in my opinion.

It may not be fun, but it's *fair* if the rules are the same for everyone, and clear ahead of time.

Also note that this revision doesn't prevent writing extra moves between the move slots on the FMC sheet, as long as they are written in line without extra symbols. (I don't think that's an important thing to prevent, so long as the Regulations themselves don't specify how to fill out the official FMC sheets.)

**Comment by lgarron** [comment]
Anyhow, since the two opinions here so far are against requiring an official sheet: what about allowing a separate page as long as it is blank apart from name/WCA ID/round info and the solution?

And if you think that's not a solution either, do you think what I proposed here about boxing/circling solutions is good?

**Comment by Laura-O** [comment]
> First image: T/F is an ambiguous symbol, R is not written in line, and the struck-through move is not clearly blacked out (although the latter is at the discretion of the Delegate).

Where is a T? :confused: 

> Do you think these cases are actually ambiguous under this revision?

Yes, 100% ambiguous.

I also don't see the reason why we should forbid such a notation. I am very certain that I could ask 100 people on the street and 99 of them would interpret the letters correctly and the 100th is a humanoid robot which didn't learn to interpret crossed out letters. :stuck_out_tongue: 

Joking aside, we have always tried to keep regulations as simple and reasonable as possible. For me, DNFing the above examples would be difficult to do consistently (by all delegates) and is unreasonable. I have checked hundreds (if not thousands) of FMC solutions and I can't remember many cases where the addition of extra moves was ambiguous. So why should this be forbidden?

**Comment by pedrosino** [comment]
I'm aligned with Laura. I also personally don't see the problem with ( ) since they have no meaning in our notation (different from [ ] which do have a meaning).

Boxing/circling may not be necessary and/or practical, if you did a lot of crossing out moves for example.
I'm also against requiring a sheet with only name and solution. As I said, it would be a kickback from what we have now, and you're right that requiring a re-write would be fair as everyone would be under the same conditions, but I don't think is necessary or helps the judge/checker.

Most problems I face with FM are wrong notation and bad hand writing.

**Comment by xsrvmy** [comment]
With the issue with parens, we need to make clear that notation such as F (U R U' R')2 F' for repeated moves is forbidden. Commutators are listed as invalid in the guidelines so this probably should be disallowed explicitly as well.

**Comment by lgarron** [comment]
> With the issue with parens, we need to make clear that notation such as F (U R U' R')2 F' for repeated moves is forbidden. Commutators are listed as invalid in the guidelines so this probably should be disallowed explicitly as well.

This current revision of this PR already includes the following examples of invalid notation:

    (U)2, (U)', (R U)'

I hope this makes it clear that (U R U' R')2 is not valid, although I suppose we could as add it to the morass if anyone thinks someone could make the opposite conclusion.

**Comment by xsrvmy** [comment]
@pedrosino How bad is the handwriting issue? There are only 12 valid characters (URFDLBwxyz'2) outside brackets and 6 valid characters (urfdlb) inside brackets and the only ones that look similar are z and 2. Even then z and 2 cannot be interchanged to form valid notation.

**Comment by Laura-O** [comment]
> @pedrosino How bad is the handwriting issue? There are only 12 valid characters (URFDLBwxyz'2) outside brackets and 6 valid characters (urfdlb) inside brackets and the only ones that look similar are z and 2. Even then z and 2 cannot be interchanged to form valid notation.

Sorry, but :woman_facepalming: 

Bad handwriting is by far the most annoying problem in FMC. You don't read WCA reports and might have never judged FMC solutions, but it should be obvious that there are way more letters that are similar when written in a sloppy way.

A few examples:
U written with a small bend on the right side -> D
B written with the middle horizontal line far away from the vertical line -> D
B not fully closed on the bottom -> R
And so on...

**Comment by pedrosino** [comment]
There's also the F that can look like a T, or R too closed that looks like a B, L too curved and looks like a U.

**Comment by xsrvmy** [comment]
I was mainly thinking about the F/T issue. T is not valid notation to begin with, so should the checker always assume that a poorly written F is an F?

Another point I want to make is that sometimes the accepted way of writing a letter can be different in different countries. For example, some Chinese judges might not be able to read a hand-written z, small f or small r.

File: wca-guidelines.md [hunk 3]
```diff
@@ -201,12 +205,16 @@ To be more informative, each Guideline is classified using one of the following
...
+- E2c2+) [EXAMPLE] Examples that are not considered a single unambiguous move sequence written out as one or multiple horizontal lines of text: a solution with extra moves, multiple solution sequences without exactly one boxed/circled solution, a solution with ambiguous symbols (e.g. a letter that looks similar to "B" and "R" but is not clearly one of them), a solution written in a spiral, a solution with stars or arrows indicating moves to be performed out of written order (e.g. "insertions" that are not written in-line with the rest of the move sequence, "pre-moves" that are not incorporated into the move sequence using valid notation), any arrangement of symbols that cannot be interpreted as an unambiguous sequence of "top-to-bottom" text lines with each line in "left-to-right" reading order.
+- E2c2++) [CLARIFICATION] The submitted page may contain written moves on both sides, as long only one side has a boxed/circled solution.
+- E2c3+) [ADDITION] A blacked-out move must be covered using ink in a rectangle, oval, or other solidly filled shape. A partially filled shape is permitted as long as it clearly blacks out the move, at the discretion of the Delegate.
```
Comment (xsrvmy) [inline wca-guidelines.md]: E2c3+) "A blacked-out MOVE must be ..."
E2c3 also includes any other symbol. Should this read "A blacked-out move or symbol ..."

File: wca-guidelines.md [hunk 4]
```diff
@@ -201,12 +205,16 @@ To be more informative, each Guideline is classified using one of the following
...
+- E2c2+) [EXAMPLE] Examples that are not considered a single unambiguous move sequence written out as one or multiple horizontal lines of text: a solution with extra moves, multiple solution sequences without exactly one boxed/circled solution, a solution with ambiguous symbols (e.g. a letter that looks similar to "B" and "R" but is not clearly one of them), a solution written in a spiral, a solution with stars or arrows indicating moves to be performed out of written order (e.g. "insertions" that are not written in-line with the rest of the move sequence, "pre-moves" that are not incorporated into the move sequence using valid notation), any arrangement of symbols that cannot be interpreted as an unambiguous sequence of "top-to-bottom" text lines with each line in "left-to-right" reading order.
+- E2c2++) [CLARIFICATION] The submitted page may contain written moves on both sides, as long only one side has a boxed/circled solution.
```
Comment (xsrvmy) [inline wca-guidelines.md]: E2c2++) Should this read "as long AS only one side..."?

**Comment by xsrvmy** [comment]
Also, a dot is listed as a disallowed symbol. Does that include stray pen marks that cannot be confused with a quote mark (eg. putting a dot on the page by habit or by accident)?

File: wca-guidelines.md (hunk 4 above)
Comment (lgarron) [reply wca-guidelines.md]: Yep, thanks!

File: wca-guidelines.md (hunk 3 above)
Comment (lgarron) [reply wca-guidelines.md]: Yeah, good idea. Done.

**Comment by lgarron** [comment]
> Also, a dot is listed as a disallowed symbol. Does that include stray pen marks that cannot be confused with a quote mark (eg. putting a dot on the page by habit or by accident)?

Yep. If you have a better idea that can be judged consistently, I'm all ears, though.

**Comment by xsrvmy** [comment]
@lgarron Is it appropriate to ignore any inline symbols that are clearly invalid? Alternatively, can a dot be considered a solidly filled shape?

**Comment by lgarron** [comment]
Closed in favor of #678.

**Pull request closed without merging**